sdk/goroutine.go

```diff
@@ -23,58 +24,113 @@ import (
 	"github.com/pkg/errors"
 )
 
+type GoRoutine struct {
+	ctx     context.Context
+	Name    string
+	Func    func(ctx context.Context)
+	Restart bool
+	Active  bool
+}
+
 // GoRoutines contains list of routines that have to stay up
 type GoRoutines struct {
-	mutex  sync.Mutex
-	status map[string]bool
+	status []*GoRoutine
```

Comment:
Should we keep the mutex to protect the status slice ? In case Run is called by multiple routines ?

Reply:
done